src/Utopia/Messaging/Adapter/Email/Mailgun.php

```diff
@@ -61,6 +62,19 @@ protected function process(Email $message): string
             ]),
         );
 
-        return $response;
+        $statusCode = $result['statusCode'];
+
+        if ($statusCode >= 200 && $statusCode < 300) {
+            $response->setDeliveredTo(\count($message->getTo()));
+            foreach ($message->getTo() as $to) {
+                $response->addToDetails($to);
```

Comment:
hmm, details hold recipient and error message for each recipient. So it doesn't really make sense to pass the whole array of recipients.

Reply:
It would make sense if the method then iterated each item of the array instead of doing it at the call site, at least for cases where we want to add success for all of them